src/app/cli.cljs

```diff
@@ -5,34 +5,72 @@
             ["child_process" :as cp]
             [clojure.string :as string]
             [cljs.reader :refer [read-string]]
-            [clojure.string :as string]))
+            [clojure.string :as string]
+            [cljs.core.async :refer [put! chan <! >! timeout close!]])
+  (:require-macros [cljs.core.async.macros :refer [go go-loop]]))
+
+(defn get-chan-file [x]
+  (let [chan-file (chan)]
+    (fs/readFile
+     x
+     "utf8"
+     (fn [err content]
+       (when (some? err) (.error js/console "Error reading file:" err))
+       (go (>! chan-file (if (some? err) nil content)))))
```

Comment:
Never create a `go` block just to put something on a channel. Also note that it is invalid to put `nil` on a channel.

```
(async/put! chan-file (if (some? err) ::err content))
```

It is also bad to create a one-use channel for this. Instead pass in a channel so that all file contents can go into one channel directly instead of you batching them up again later.

Reply:
`::err`? namespaced keyword, why?

I guess I'm using channels like using `Promise.all` in this file. Do you have an example for such cases?

Reply:
You can't put `nil` on a channel so you put some other unique enough value that you can check when reading from the channel.

Reply:
> It is also bad to create a one-use channel for this.

Tried but I still found code written with one-use channel is more concise. `(timeout 1000)`  is an "one-use" channel, right?